Formula/nlopt.rb

```diff
@@ -13,81 +13,135 @@ class Nlopt < Formula
     end
   end
 
   test do
-    # Based on https://nlopt.readthedocs.io/en/latest/NLopt_Tutorial/#Example_in_C.2FC.2B.2B
-    (testpath/"test.c").write <<~EOS
+    # https://nlopt.readthedocs.io/en/latest/NLopt_Tutorial/#example-in-cc
+    (testpath/"tutorial.c").write <<~EOS
       #include <math.h>
-      #include <nlopt.h>
       #include <stdio.h>
-      double myfunc(unsigned n, const double *x, double *grad, void *my_func_data) {
+      #include <nlopt.h>
+      double myfunc(unsigned n, const double *x, double *grad,
+                    void *my_func_data) {
         if (grad) {
-          grad[0] = 0.0;
-          grad[1] = 0.5 / sqrt(x[1]);
+          grad[0] = 0.0; grad[1] = 0.5 / sqrt(x[1]);
         }
         return sqrt(x[1]);
       }
       typedef struct { double a, b; } my_constraint_data;
-      double myconstraint(unsigned n, const double *x, double *grad, void *data) {
+      double myconstraint(unsigned n, const double *x, double *grad,
+                          void *data) {
         my_constraint_data *d = (my_constraint_data *) data;
         double a = d->a, b = d->b;
         if (grad) {
-          grad[0] = 3 * a * (a*x[0] + b) * (a*x[0] + b);
-          grad[1] = -1.0;
+          grad[0] = 3 * a * (a * x[0] + b) * (a * x[0] + b); grad[1] = -1.0;
         }
-        return ((a*x[0] + b) * (a*x[0] + b) * (a*x[0] + b) - x[1]);
-       }
-       int main() {
-        double lb[2] = { -HUGE_VAL, 0 }; /* lower bounds */
+        return (a * x[0] + b) * (a * x[0] + b) * (a * x[0] + b) - x[1];
+      }
+      int main() {
+        double lb[2] = { -HUGE_VAL, 0.0 };
         nlopt_opt opt;
-        opt = nlopt_create(NLOPT_LD_MMA, 2); /* algorithm and dimensionality */
+        opt = nlopt_create(NLOPT_LD_MMA, 2);
         nlopt_set_lower_bounds(opt, lb);
-        nlopt_set_min_objective(opt, myfunc, NULL);
-        my_constraint_data data[2] = { {2,0}, {-1,1} };
+        nlopt_set_min_objective(opt, myfunc, 0);
+        my_constraint_data data[2] = { { 2.0, 0.0 }, { -1.0, 1.0 } };
         nlopt_add_inequality_constraint(opt, myconstraint, &data[0], 1e-8);
         nlopt_add_inequality_constraint(opt, myconstraint, &data[1], 1e-8);
         nlopt_set_xtol_rel(opt, 1e-4);
-        double x[2] = { 1.234, 5.678 };  /* some initial guess */
-        double minf; /* the minimum objective value, upon return */
-
-        if (nlopt_optimize(opt, x, &minf) < 0)
+        double x[2] = { 1.234, 5.678 };
+        double minf;
+        if (nlopt_optimize(opt, x, &minf) < 0) {
+          fprintf(stderr, "nlopt failed!");
+          nlopt_destroy(opt);
           return 1;
-        else
-          printf("found minimum at f(%g,%g) = %0.10g", x[0], x[1], minf);
+        }
+        printf("found minimum at f(%g,%g) = %0.10g", x[0], x[1], minf);
         nlopt_destroy(opt);
+        return 0;
+      }
+    EOS
+    # https://nlopt.readthedocs.io/en/latest/NLopt_Tutorial/#example-in-c
+    (testpath/"tutorial.cpp").write <<~EOS
```

Comment:
Any chance some of these files are distributed with the NLopt source code (or a similar example test) that we could copy into the `share` directory rather than writing out these big, long C and C++ programs? It would be nice from a maintenance perspective to have less code here.

Reply:
(and hope that upstream would keep examples up to date, and working...)